app_flutter/lib/widgets/task_icon.dart

```diff
@@ -25,22 +25,46 @@ class TaskIcon extends StatelessWidget {
   /// A lookup table for matching [stageName] to [Image].
   ///
   /// [stageName] is based on the backend.
-  static final Map<String, Image> stageIcons = <String, Image>{
-    StageName.cirrus: Image.asset('assets/cirrus.png'),
-    StageName.luci: Image.asset('assets/chromium.png'),
-    StageName.devicelab: Image.asset('assets/android.png'),
-    StageName.devicelabWin: Image.asset('assets/windows.png'),
-    StageName.devicelabIOs: Image.asset('assets/apple.png'),
-  };
+  Widget stageIconForBrightness(Brightness brightness) {
+    final Color blendFilter = brightness == Brightness.dark ? Colors.white : null;
 
-  @override
-  Widget build(BuildContext context) {
-    Widget icon = const Icon(Icons.help);
     if (qualifiedTask.stage == StageName.luci && qualifiedTask.task == 'linux_bot') {
-      icon = Image.asset('assets/fuchsia.png');
-    } else if (stageIcons.containsKey(qualifiedTask.stage)) {
-      icon = stageIcons[qualifiedTask.stage];
+      return Image.asset(
+        'assets/fuchsia.png',
+      );
     }
+    switch (qualifiedTask.stage) {
+      case StageName.cirrus:
+        return Image.asset(
+          'assets/cirrus.png',
+          color: blendFilter,
```

Comment:
Do you know if the DOM implementation is going to get this support? I lean towards duplicating the assets if this only works in CanvasKit since the DOM was the quicker version.

Reply:
I'm not sure. Whatever is currently up on flutter-dashboard.appspot.com is still using CanvasKit for sure though.

Reply:
This was implemented (at least partially) in the HTML backend last week: https://github.com/flutter/engine/commit/a0983d36f7bf2b4f45a5762ce40bb81d8a3f8f9f.